lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
Sorry that `[ ]` threw me off... It's just meant to be optional, not an array.

So we have:
```js
.publish('message', cb); // one message, no labels

.publish('message', ['label1', 'label2'], cb); // one message with labels

.publish([ { data: 'message1' }, { data: 'message2' } ], cb); // multiple messages, no labels.

.publish([{
  data: 'message1'
  labels: ['label1', 'label2']
}, {
 data: 'message2'
 labels: ['label3', 'label4']
}], cb); // multiple messages, with labels
```